packages/concerto-core/lib/introspect/parser.pegjs

```diff
@@ -833,7 +844,7 @@ Decorators
   = decorators:(d:Decorator __ {return d;})*
 
 AssetDeclaration
-  = decorators:Decorators __ abstract:AbstractToken? __ AssetToken __ id:Identifier __ idField:IdentifiedByField? __ classExtension: ClassExtension? __
```

Comment:
I thought the design was to always have asset/participant/transaction/events identified, so maybe we disallow it in the grammar?

Reply:
I think what we should ensure is that in a type hierarchy the first non-abstract asset (etc) has an identifier declared. I will tighten up the programmatic validation and add tests for that.